tests/foreman/api/test_environment.py

```diff
@@ -24,44 +24,54 @@
 from requests.exceptions import HTTPError
 
 from robottelo.api.utils import one_to_many_names
-from robottelo.datafactory import filtered_datapoint
+from robottelo.datafactory import invalid_environments_list
 from robottelo.datafactory import invalid_names_list
-from robottelo.test import APITestCase
+from robottelo.datafactory import parametrized
+from robottelo.datafactory import valid_environments_list
 
 
-@filtered_datapoint
-def valid_data_list():
-    """Return a list of various kinds of valid strings for Environment entity"""
-    return [gen_string('alpha'), gen_string('numeric'), gen_string('alphanumeric')]
+@pytest.fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
 
 
-class EnvironmentTestCase(APITestCase):
-    """Tests for environments."""
+@pytest.fixture(scope='module')
+def module_loc():
+    return entities.Location().create()
+
+
+@pytest.fixture(scope='module')
+def module_env():
+    return entities.Environment().create()
+
 
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.org = entities.Organization().create()
-        cls.loc = entities.Location().create()
+@pytest.fixture(scope='module')
+def module_env_attrs(module_env):
+    return set(module_env.update_json([]).keys())
```

Comment:
This doesn't really appear to be related to setup, I think it would be better to just do this lookup within the tests that need it. Its a bit of repeated code, but its a simple statement.

Reply:
ACK, moved.